PR: https://github.com/odoo/odoo/pull/135932

From: 2b28009f5b3f342b8923bfce9cb6858adb1d99c9
From: Lucas Perais
Date: 2023-10-20 09:35:12

Structural Changes: 5
Total Changes: 30

[IMP] base_automation: better UX

This commit improves the general feel when creating or displaying base automations
The improvement relies mainly on correctly labelling the fields

task-id-3450200

closes odoo/odoo#135932

Related: odoo/enterprise#47609
Signed-off-by: Lucas Perais (lpe) <lpe@odoo.com>

================================= pseudo patch: =================================

--- a/odoo/addons/base/models/ir_actions.py
+++ b/odoo/addons/base/models/ir_actions.py
@@ -508,7 +508,7 @@ class IrActionsServer(models.Model):
     groups_id = fields.Many2many('res.groups', 'ir_act_server_group_rel',
                                  'act_id', 'gid', string='Allowed Groups', help='Groups that can execute the server action. Leave empty to allow everybody.')
 
-    update_field_id = fields.Many2one('ir.model.fields', string='Field to update', default=_default_update_field_id, ondelete='cascade')
+    update_field_id = fields.Many2one('ir.model.fields', string='Field to Update', default=_default_update_field_id, ondelete='cascade')
     update_related_model_id = fields.Many2one('ir.model', compute='_compute_update_related_model_id')
 
     value = fields.Text(help="For Python expressions, this field may hold a Python expression "
@@ -518,12 +518,12 @@ class IrActionsServer(models.Model):
                              "For Static values, the value will be used directly without evaluation, e.g."
                              "`42` or `My custom name` or the selected record.")
     evaluation_type = fields.Selection([
-        ('value', 'Static value'),
+        ('value', 'Custom Value'),
         ('equation', 'Python expression')
     ], 'Value Type', default='value', change_default=True)
     resource_ref = fields.Reference(
         string='Record', selection='_selection_target_model', inverse='_set_resource_ref')
-    selection_value = fields.Many2one('ir.model.fields.selection', string="Selection value", ondelete='cascade',
+    selection_value = fields.Many2one('ir.model.fields.selection', string="Custom Value", ondelete='cascade',
                                       domain='[("field_id", "=", update_field_id)]', inverse='_set_selection_value')
 
     value_field_to_show = fields.Selection([

--- a/odoo/addons/base/models/ir_model.py
+++ b/odoo/addons/base/models/ir_model.py
@@ -1059,8 +1059,10 @@ class IrModelFields(models.Model):
 
     @api.depends('field_description', 'model')
     def _compute_display_name(self):
+        IrModel = self.env["ir.model"]
         for field in self:
-            field.display_name = f'{field.field_description} ({field.model})'
+            model_string = IrModel._get(field.model).name
+            field.display_name = f'{field.field_description} ({model_string})'
 
     def _reflect_field_params(self, field, model_id):
         """ Return the values to write to the database for the given field. """

--- a/odoo/addons/base/views/ir_actions_views.xml
+++ b/odoo/addons/base/views/ir_actions_views.xml
@@ -333,26 +333,26 @@
                             <group name="action_content">
                                 <field name="model_name" invisible="1"/>
                                 <field name="available_model_ids" invisible="1"/>
-                                <field name="model_id" options="{'no_create': True}" groups="base.group_no_one"
+                                <field name="model_id" options="{'no_create': True, 'no_open': True}" groups="base.group_no_one"
                                        domain="[('id', 'in', available_model_ids)]"/>
-                                <field name="model_id" options="{'no_create': True}" groups="!base.group_no_one" invisible="context.get('default_model_id')"
+                                <field name="model_id" options="{'no_create': True, 'no_open': True}" groups="!base.group_no_one" invisible="context.get('default_model_id')"
                                        domain="[('id', 'in', available_model_ids)]"/>
                                 <field name="groups_id" widget="many2many_tags" groups="base.group_no_one" />
                                 <field name="state" string="Action Type"/>
                                 <field name="update_field_id" invisible="state != 'object_write'"
-                                    options="{'no_create': True}"
+                                    options="{'no_create': True, 'no_open': True}"
                                     domain="['|', ('model_id', '=', crud_model_id), ('model_id', '=', model_id)]"/>
-                                <field name="evaluation_type" invisible="state != 'object_write'"/>
+                                <field name="evaluation_type" invisible="state != 'object_write'" widget="radio" options="{'horizontal': True}"/>
                                 <field name="value_field_to_show" invisible="1"/>
                                 <field name="update_related_model_id" invisible="1"/>
-                                <field name="value" invisible="state != 'object_write' or value_field_to_show != 'value' or evaluation_type != 'value'"/>
-                                <field name="value" widget="code" options="{'mode': 'python'}" invisible="state != 'object_write' or evaluation_type != 'equation'"/>
-                                <field name="resource_ref" string="Value" options="{'model_field': 'update_related_model_id', 'no_create': True}" invisible="state != 'object_write' or value_field_to_show != 'resource_ref' or evaluation_type == 'equation'"/>
-                                <field name="selection_value" options="{'no_create': True}" invisible="state != 'object_write' or value_field_to_show != 'selection_value' or evaluation_type == 'equation'"/>
+                                <field name="value" invisible="state != 'object_write' or value_field_to_show != 'value' or evaluation_type != 'value'" string="Custom Value"/>
+                                <field name="value" widget="code" options="{'mode': 'python'}" invisible="state != 'object_write' or evaluation_type != 'equation'" string="Expression"/>
+                                <field name="resource_ref" string="Custom Value" options="{'model_field': 'update_related_model_id', 'no_create': True, 'no_open': True}" invisible="state != 'object_write' or value_field_to_show != 'resource_ref' or evaluation_type == 'equation'"/>
+                                <field name="selection_value" options="{'no_create': True, 'no_open': True}" invisible="state != 'object_write' or value_field_to_show != 'selection_value' or evaluation_type == 'equation'"/>
 
                                 <field name="type" invisible="1"/>
                                 <field name="crud_model_id"
-                                    options="{'no_create': True}"
+                                    options="{'no_create': True, 'no_open': True}"
                                     invisible="state != 'object_create'"
                                     required="state == 'object_create'"/>
                                 <field name="crud_model_name" invisible="1"/>
@@ -360,7 +360,7 @@
                                 <field name="link_field_id"
                                     domain="[('model_id', '=', model_id), ('relation', '=', crud_model_name),
                                     ('ttype', 'in', ['many2one', 'one2many', 'many2many'])]"
-                                    options="{'no_create': True}"
+                                    options="{'no_create': True, 'no_open': True}"
                                     invisible="state != 'object_create'"
                                     context="{'default_model_id': model_id, 'default_relation': crud_model_name}"/>
                             </group>
